### PR TITLE
[exporter/clickhouse] Add internal component metrics

### PR DESCRIPTION
**Pull request by Frapschen**
**Description:** <Describe what has changed.>

For the added metrics please see documentation.md in this PR.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/17559

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: exporter/clickhouseexporter/metadata.yaml [hunk 1]
```diff
@@ -12,3 +12,62 @@ status:
...
+telemetry:
+  metrics:
+    exporter_clickhouse_sent_spans:
+      enabled: true
+      description: Number of spans the exporter sent.
+      unit: "{spans}"
+      sum:
+        value_type: int
+        monotonic: true
+      attributes: [error]
```
Comment (atoulme) [inline exporter/clickhouseexporter/metadata.yaml:25]: where is the attribute error defined?

File: exporter/clickhouseexporter/metadata.yaml [hunk 2]
```diff
@@ -12,3 +12,62 @@ status:
...
+        bucket_boundaries: [5, 10, 20, 50, 100, 200, 500, 1000, 2000, 5000]
+      attributes: [error]
+    exporter_clickhouse_sent_batch_size:
+      enabled: true
+      description: How many count of a batch sent to Clickhouse. Clickhouse recommend inserting data in packets of at least 1000 rows, or no more than a single request per second.
+      unit: int
+      histogram:
+        value_type: int
+        bucket_boundaries: [100, 250, 500, 750, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000, 20000, 30000, 50000, 100000]
+      attributes: [telemetry_type]
```
Comment (atoulme) [inline exporter/clickhouseexporter/metadata.yaml:73]: Why do you define metrics by signal type everywhere but for this one, and use an attribute for telemetry type instead? Where is this attribute defined?

**Review by SpencerTorres — APPROVED** [review]
Agree with @atoulme's notes, but overall good addition. Thanks for adding this

**Comment by Frapschen** [comment]
Wait this PR https://github.com/open-telemetry/opentelemetry-collector/pull/10942

File: exporter/clickhouseexporter/documentation.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
+[comment]: <> (Code generated by mdatagen. DO NOT EDIT.)
+
+# clickhouse
+
+## Internal Telemetry
+
+The following telemetry is emitted by this component.
+
+### otelcol_exporter_clickhouse_sent_logs
```
Comment (dmitryax) [inline exporter/clickhouseexporter/documentation.md:9]: There is overlap with already reported metrics https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/metadata.yaml#L68. Please ensure that we don't have any duplicates

**Pull request closed without merging**